src/arithmetic/algebraic_expression.h

```diff
@@ -42,53 +46,30 @@ struct AlgebraicExpressionNode {
     AlgebraicExpressionNodeType type;
 };
 
-AlgebraicExpressionNode *AlgebraicExpressionNode_NewOperationNode(AL_EXP_OP op);
-AlgebraicExpressionNode *AlgebraicExpressionNode_NewOperandNode(GrB_Matrix operand);
+typedef struct {
+  AlgebraicExpressionNode *exp_root;
+  AlgebraicExpressionNode *exp_leaf;
```

Comment:
An expression might have several leaf nodes

Reply:
Yeah, bad name - this is the right-hand 'tail' of the tree, so that we can append nodes without traversing from the root.